sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOperation.cs

```diff
@@ -80,9 +87,15 @@ internal RecognizeCustomFormsOperation(ServiceClient operations, ClientDiagnosti
         /// <param name="client">The client used to check for completion.</param>
         public RecognizeCustomFormsOperation(string operationId, FormRecognizerClient client)
         {
-            Id = operationId;
             _serviceClient = client.ServiceClient;
             _diagnostics = client.Diagnostics;
+
+            string[] substrs = operationId.Split('/');
+
+            _resultId = substrs.Last();
+            _modelId = substrs.First();
```

Comment:
Same here re: magic numbers.

Reply:
It would be good to flag this as an area for greater input validation as well... maybe we want to do guid format checking on _resultId and _modelId here, rather than blindly passing them to a service endpoint?

Reply:
I'll add a comment to the code so we don't forget about it. That should be part of our argument validation issue (#10385). We could already store the private fields as GUIDs rather than keeping them as strings.